solution.cpp

```diff
@@ -1,138 +1,134 @@
- * - `outfilename`: A string representing the file path to write to.
- *
- * Returns:
- * - 0 on success, or -1 on error.
- */
+int find_max(const string& infilename);
+int write_primes(const int size,
+                 bool primes[],
+                 const string& infilename,
+                 const string& outfilename);
 int write_composites( const int size,
                       bool primes[],
                       const string& infilename,
                       const string& outfilename );
 
-// ----------------------------------------------------------------------------
-
-int main() {
-    int max = find_max("input.txt");
-    if (max == -1) {
-        cout << "ERROR in `find_max()`" << endl;
+int main()
+{
+	int max = find_max("input.txt");
+	if (max == -1) {
+        cout << "ERROR in 'find_max()'" << endl;
         return 1;  // error
     }
-
+    
     // - Runtime sized arrays are a C99 feature, and a GCC and Clang extension.
     //   Thus `bool primes[max+1];` is not standards compliant.  It should work
     //   on Xcode (which uses Clang) and in Linux (where most people use either
     //   GCC or Clang), but Visual Studio appears not to allow it.
     bool * primes = (bool *) malloc( sizeof(bool) * (max+1) );
     sieve(max+1, primes);
-
-    int ret;  // for storing return values, to check for error codes
-    //
-    ret = write_primes(max+1, primes, "input.txt", "primes.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_primes()`" << endl;
-        return 1;  // error
-    }
-    //
-    ret = write_composites(max+1, primes, "input.txt", "composites.txt");
+    
+	int ret;							// for storing return values, to check for error codes
+	ret = write_primes(max + 1, primes, "input.txt", "primes.txt");
+	if (ret == -1) {
+		cout << "ERROR in `write_primes()`" << endl;
+		return 1; // error
+	}
+
+	ret = write_composites(max+1, primes, "input.txt", "composites.txt");
     if (ret == -1) {
         cout << "ERROR in `write_composites()`" << endl;
         return 1;  // error
     }
 
-    return 0;  // success
+		return 0;
 }
 
-// ----------------------------------------------------------------------------
+int find_max(const string& infilename)
+{
+    int max = 0,
+        x;
+    ifstream inputFile(infilename);
+    if (!inputFile.is_open())
+        return -1;									//error
+    while (inputFile >> x)
+    {
+        if (x > max)
+            max = x;
+    }
+    return max;
+}
 
-// TODO: write prototyped functions
+void sieve(const int size, bool primes[])
+{
+    primes[0] = false;        //0 is not prime
+    primes[1] = false;        //1 is not prime
+    primes[2] = true;         //2 is the smallest prime
+    for (int i = 3; i < size; i++)              //calculates primes and stores them as either true or false in the array
+    {
+        for (int j = 2; j < i; j++)
+        {
+            if (i%j == 0)
+                primes[i] = false;
+        }
+    }
```

Comment:
this is not the sieve of Eratosthenes algorithm, and is actually far less efficient -- but it looks like it'll accomplish the same thing